android/src/main/java/com/swmansion/rnscreens/ScreenContainer.java

```diff
@@ -235,8 +235,12 @@ private void detachScreen(ScreenFragment screenFragment) {
     getOrCreateTransaction().remove(screenFragment);
   }
 
+  protected int getScreenActive(ScreenFragment screenFragment) {
+    return screenFragment.getScreen().getActive();
+  }
+
   protected boolean isScreenActive(ScreenFragment screenFragment) {
-    return screenFragment.getScreen().isActive();
+    return getScreenActive(screenFragment) != 0;
```

Comment:
it looks like after this change `active` changes its semantic. We should avoid calling it the same way when thinking of it as meaning different things. In particular you can get `active` as an int and also get active as boolean. This suggest that active has different meanins and we should use different naming for these two things